[Bug]: Compilation issues with GCC 15 and Python 3.14 POSIX_C_SOURCE redefined

### Version / commit SHA

v4.0.0RC2

### Environment

OS: Arch Linux
Compiler: GCC 15
Build type: Debug
Cmake flags: -DCMAKE_CXX_COMPILER=/usr/bin/g++-15

The mishandling of POSIX_C_SOURCE and XOPEN_SOURCE is a very long standing issue on Python: https://github.com/python/cpython/issues/61322

Generally the proposed solution is to make it the first include: https://github.com/astropy/astropy/pull/19340

### Minimal reproducer

```
mkdir build
cd build
cmake -DCMAKE_CXX_COMPILER=/usr/bin/g++-15 ..
cmake --build . -- VERBOSE=1  -j 6
```

### Expected behaviour

It compiles

### Actual behaviour

```
[100%] Built target qa_KnownSharedLibBlocks
In file included from /usr/include/python3.14/Python.h:14,
                 from /home/lukken/projects/contribute/gnuradio-sdr/gnuradio4/blocks/basic/include/gnuradio-4.0/basic/PythonInterpreter.hpp:24,
                 from /home/lukken/projects/contribute/gnuradio-sdr/gnuradio4/blocks/basic/include/gnuradio-4.0/basic/PythonBlock.hpp:4,
                 from /home/lukken/projects/contribute/gnuradio-sdr/gnuradio4/blocks/basic/test/qa_PythonBlock.cpp:5:
/usr/include/python3.14/pyconfig.h:2007:9: error: ‘_POSIX_C_SOURCE’ redefined [-Werror]
 2007 | #define _POSIX_C_SOURCE 200809L
      |         ^~~~~~~~~~~~~~~
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/15.2.1/include/c++/x86_64-pc-linux-gnu/bits/os_defines.h:39,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/15.2.1/include/c++/x86_64-pc-linux-gnu/bits/c++config.h:727,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/15.2.1/include/c++/bits/version.h:51,
                 from /usr/lib/gcc/x86_64-pc-linux-gnu/15.2.1/include/c++/version:38,
                 from /home/lukken/projects/contribute/gnuradio-sdr/gnuradio4/build/_deps/ut-src/include/boost/ut.hpp:16,
                 from /home/lukken/projects/contribute/gnuradio-sdr/gnuradio4/blocks/basic/test/qa_PythonBlock.cpp:2:
/usr/include/features.h:319:10: note: this is the location of the previous definition
  319 | # define _POSIX_C_SOURCE        202405L
      |          ^~~~~~~~~~~~~~~
/usr/include/python3.14/pyconfig.h:2043:9: error: ‘_XOPEN_SOURCE’ redefined [-Werror]
 2043 | #define _XOPEN_SOURCE 700
      |         ^~~~~~~~~~~~~
/usr/include/features.h:234:10: note: this is the location of the previous definition
  234 | # define _XOPEN_SOURCE  800
      |          ^~~~~~~~~~~~~
cc1plus: note: unrecognized command-line option ‘-Wno-unused-command-line-argument’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
```

### Logs / backtrace

```shell

```

--- a/blocks/basic/include/gnuradio-4.0/basic/PythonInterpreter.hpp
+++ b/blocks/basic/include/gnuradio-4.0/basic/PythonInterpreter.hpp
@@ -24,15 +24,6 @@
 
 #include <gnuradio-4.0/Message.hpp>
 
-#ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wold-style-cast"
-#ifndef __clang__
-#pragma GCC diagnostic ignored "-Wuseless-cast"
-#endif
-#endif
-#include <Python.h>
-
 #include <numpy/numpyconfig.h>
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
 #include <numpy/arrayobject.h>

--- a/blocks/basic/include/gnuradio-4.0/basic/PythonInterpreter.hpp
+++ b/blocks/basic/include/gnuradio-4.0/basic/PythonInterpreter.hpp
@@ -1,6 +1,16 @@
 #ifndef GNURADIO_PYTHONINTERPRETER_HPP
 #define GNURADIO_PYTHONINTERPRETER_HPP
 
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+#ifndef __clang__
+#pragma GCC diagnostic ignored "-Wuseless-cast"
+#endif
+#endif
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
 #include <atomic>
 #include <cassert>
 #include <cctype>

--- a/blocks/basic/test/qa_PythonBlock.cpp
+++ b/blocks/basic/test/qa_PythonBlock.cpp
@@ -1,7 +1,9 @@
+
+#include <gnuradio-4.0/basic/PythonBlock.hpp>
+
 #include <boost/ut.hpp>
 
 #include <gnuradio-4.0/Graph.hpp>
-#include <gnuradio-4.0/basic/PythonBlock.hpp>
 
 #include <gnuradio-4.0/Scheduler.hpp>
 #include <gnuradio-4.0/meta/UnitTestHelper.hpp>

